modules/date-picker/js/date-picker_directive.js

```diff
@@ -123,7 +123,7 @@
 
             timer2 = $timeout(function()
             {
-                var yearSelector = $element.find('.lx-date-picker__year-selector');
+                var yearSelector = $document.find('.lx-date-picker__year-selector');
```

Comment:
Yes, more efficient solution! I'll test this asap and change the code accordingly.